### DISCUSSION TITLE
Fail2ban for IMAP/SMTP

### DISCUSSION THREAD

**Discussion by vitobotta** [Mailu/Mailu#3079] [User support]
Hi, I set up fail2ban as described in the docs but when I tried a buteforce (using a vpn), the failed login message in the log didn't match the regexes in the two filters described in the docs. I had to create another filter like this:

```
[Definition]
failregex = ^.* Login failed for .* from <HOST>.$
ignoreregex =
journalmatch = CONTAINER_TAG=mailu-admin
```

This seems to trigger a ban as expected. Is the documentation not up to date about this topic or what am I missing?

Also, I want to secure IMAP and SMTP so I added these two filters found with some searching:

```
[Definition]
failregex = (?: pop3-login|imap-login): .*(?:Authentication failure|Aborted login \(auth failed|Aborted login \(tried to use disabled|Disconnected \(auth failed|Aborted login \(\d+ authentication attempts).*rip=`<HOST>`
ignoreregex =
journalmatch = CONTAINER_TAG=mailu-imap

[Definition]
failregex   = NOQUEUE: reject: RCPT from \S+\[<HOST>\]: 554$
              warning: (?:.*\[<HOST>\]\:.*) SASL (PLAIN|LOGIN) authentication failed: .*$
              warning: <HOST>: address not listed for hostname .*$
              status=deferred \(connect to (?:.*\[<HOST>\]\:.*)(?:\d.\:.*) Connection refused\)$
              status=deferred \(host (?:.*\[<HOST>\] said\:.*\:) Relay access denied \(.*RCPT TO.*\)\)$
              lost connection with (?:.*\[<HOST>\].*) while sending RCPT TO$
              status=deferred \(delivery temporarily suspended\: connect to (?:.*\[<HOST>\]\:.*)(?:\d.\:.*) Connection timed out\)$
              warning: non-SMTP command from (?:.*\[<HOST>\]\:.*) From: .*$
ignoreregex =
journalmatch = CONTAINER_TAG=mailu-smtp
```

Are these correct for Mailu's setup? 

Thanks

**Comment by nextgens** [comment]
> the failed login message in the log didn't match the regexes in the two filters described in the docs

Can you provide logs please?

> Are these correct for Mailu's setup?

Definitely not; with them you may end up blocking legitimate traffic. You should stick to what is documented.

**Reply by vitobotta** [reply]
>  Can you provide logs please?

The log entries when the login fails are in the format "WARNING in base: Login failed for Hhhh@fgg.com from 93.106.12.76." so it doesn't match the regexes in the docs, which expect the failed login message to include "client login failed" or "Authentication attempt from...."

> Definitely not; with them you may end up blocking legitimate traffic. You should stick to what is documented.

The documentation only mentions fail2ban config for the web app, not for imap and smtp - I am looking at https://mailu.io/2.0/faq.html. I would like to protect those too. Any suggestions? Thanks

**Reply by nextgens** [reply]
All the protocols are centralised; the documented regexp will catch all of it.

Just ensure you use the regexp that corresponds to the version you have installed. If you run master you want to follow the documentation for master

**Reply by vitobotta** [reply]
I see, I guess I misunderstood the docs. Will follow the page on master then. Thanks!
